lib/internal/webstreams/adapters.js

```diff
@@ -120,6 +121,10 @@ function newWritableStreamFromStreamWritable(streamWritable) {
   }
 
   const cleanup = finished(streamWritable, (error) => {
+    if (error?.code === 'ERR_STREAM_PREMATURE_CLOSE') {
+      error = new AbortError();
```

Comment:
`cause` proposal would make total sense here IMHO, wdyt?
```suggestion
      const err = new AbortError();
      err.cause = error;
      error = err;
```